### DISCUSSION TITLE
I want to generate the totp key with Kratos Identities in my application ,what should I do?Is theres any API or Url for this in Kratos

### DISCUSSION THREAD

**Discussion by Answerlord** [ory/kratos#2510] [Q&A] [answered]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/kratos/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/kratos/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Cloud](https://www.ory.sh/) project.
- [ ] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [ ] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe your problem

I'd like to design a RESTful API for my application to generate the totp key,like the [kratos-selfservice-ui-node](https://github.com/ory/kratos-selfservice-ui-node),so that I can accomplish the MFA.
I have found the function for generating the totp key in the code of Kratos:
```
func NewKey(ctx context.Context, accountName string, d interface {
	config.Provider
}) (*otp.Key, error) {
                ...
		return key, err
}
```
But I don't know how to use it (I don't know how to initialize the arguments that calls d),and the code of Kratos only provide the example for testing.I have no idea for using it in my code.

### Describe your ideal solution

Is there an API or Url for generating the totp key in Kratos client? Or would u mind to provide this feature in your SDKs.

### Workarounds or alternatives

I have no idea now.Maybe I can learn REACT to figure out how it works in [kratos-selfservice-ui-node](https://github.com/ory/kratos-selfservice-ui-node),and use its way to accomplish the generation of totp key

### Version

v0.9.0-alpha.2

**Comment by aeneasr** [comment, marked as answer]
Use: https://github.com/pquerna/otp
